### PR TITLE
Complete rework of the Russian localization

### PR DESCRIPTION
**Pull request by zotttttttt**
# Issue
#2592
The JSON in ru-RU.json is still valid, the structure is the same as before, only the content has been changed (for the better).

## Tasklist

 - [x] Add tests
 - [x] Add #fixes with the issue number that this PR addresses
 - [x] Generally use squash merge to rebase and clean comments before merging
 - [x] Update the [changelog](CHANGELOG.md)

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

**Review by danpaz — APPROVED** [review]
Thank you very much @zotttttttt ! 

/cc @yuryleb who was also looking at the Russian localization and may have some input here.

File: locales/ru-RU.json [hunk 1]
```diff
@@ -36,49 +36,49 @@
       "phrases": {
         "0": "Держитесь <RELATIVE_DIRECTION>.",
         "1": "Держитесь <RELATIVE_DIRECTION> по <STREET_NAMES>.",
-        "2": "Держитесь <RELATIVE_DIRECTION> по <BEGIN_STREET_NAMES>. Продолжайте по <STREET_NAMES>.",
-        "3": "Держитесь <RELATIVE_DIRECTION> чтобы остаться на <STREET_NAMES>.",
-        "4": "Bear <RELATIVE_DIRECTION> at <JUNCTION_NAME>.",
-        "5": "Bear <RELATIVE_DIRECTION> toward <TOWARD_SIGN>."
+        "2": "Держитесь <RELATIVE_DIRECTION> по <BEGIN_STREET_NAMES>. Продолжайте движение по <STREET_NAMES>.",
```
Comment (yuryleb) [inline locales/ru-RU.json:39]: "Двигайтесь <RELATIVE_DIRECTION> по <BEGIN_STREET_NAMES> на <STREET_NAMES>." ?

File: locales/ru-RU.json [hunk 2]
```diff
@@ -36,49 +36,49 @@
...
+        "3": "Держитесь <RELATIVE_DIRECTION>, чтобы остаться на <STREET_NAMES>.",
```
Comment (yuryleb) [inline locales/ru-RU.json:40]: "Двигайтесь <RELATIVE_DIRECTION> по <STREET_NAMES>." ?

File: locales/ru-RU.json [hunk 3]
```diff
@@ -36,49 +36,49 @@
...
+        "4": "Держитесь <RELATIVE_DIRECTION> по ответвлению <JUNCTION_NAME>.",
```
Comment (yuryleb) [inline locales/ru-RU.json:41]: "Двигайтесь <RELATIVE_DIRECTION> на <JUNCTION_NAME>." ?

File: locales/ru-RU.json [hunk 4]
```diff
@@ -36,49 +36,49 @@
...
+        "5": "Держитесь <RELATIVE_DIRECTION> в сторону <TOWARD_SIGN>."
```
Comment (yuryleb) [inline locales/ru-RU.json:42]: "Двигайтесь <RELATIVE_DIRECTION> в сторону <TOWARD_SIGN>." ?

File: locales/ru-RU.json [hunk 5]
```diff
@@ -132,10 +132,10 @@
     },
     "continue": {
       "phrases": {
-        "0": "Продолжайте.",
-        "1": "Продолжайте по <STREET_NAMES>.",
-        "2": "Continue at <JUNCTION_NAME>.",
-        "3": "Continue toward <TOWARD_SIGN>."
+        "0": "Продолжайте дижение.",
```
Comment (yuryleb) [inline locales/ru-RU.json:135]: Misprint in "движение". And perhaps better "Продолжите" instead of "Продолжайте"?

File: locales/ru-RU.json [hunk 6]
```diff
@@ -354,363 +354,363 @@
...
-          "Заедьте на кольцо и съезжайте на втором съезде.",
-          "Заедьте на кольцо и съезжайте на третьем съезде.",
-          "Заедьте на кольцо и съезжайте на четвертом съезде.",
-          "Заедьте на кольцо и съезжайте на пятом съезде.",
-          "Заедьте на кольцо и съезжайте на шестом съезде.",
-          "Заедьте на кольцо и съезжайте на седьмом съезде.",
-          "Заедьте на кольцо и съезжайте на восьмом съезде.",
-          "Заедьте на кольцо и съезжайте на девятом съезде.",
-          "Заедьте на кольцо и съезжайте на десятом съезде."
+          "Круговое движение. Далее проезжайте по первому съезду.",
```
Comment (yuryleb) [inline locales/ru-RU.json:450]: "Покиньте круговое движение на первом съезде" ?

**Review by yuryleb — COMMENTED** [review]
Newer Russian localization looks better than before but unfortunately it has no any relationship with "original" https://github.com/Project-OSRM/osrm-text-instructions 😞 I didn't review the whole project yet but suppose that it will be impossible to embed some pre-processing before collation of final localized string like https://github.com/Project-OSRM/osrm-text-instructions/blob/master/languages/grammar/README.md

File: locales/ru-RU.json (hunk 1 above)
Comment (zotttttttt) [reply locales/ru-RU.json:39]: That means you need to rewrite "relative_directions" as well. And "Держитесь правее" is a standard for any Russian localization of GPS voiceover,  not "Двигайтесь справа".

File: locales/ru-RU.json (hunk 2 above)
Comment (zotttttttt) [reply locales/ru-RU.json:40]: Answered in https://github.com/valhalla/valhalla/pull/2593#discussion_r488044670

I really want it to be better, but you can't just rewrite every particular Russian text with just 2 versions of "relative_directions".

File: locales/ru-RU.json (hunk 6 above)
Comment (zotttttttt) [reply locales/ru-RU.json:450]: Again, there is a particular set of ordinal directions:

`"ordinal_values": [
        "первому",
        "второму",
        "третьему",
        "четвертому",
        "пятому",
        "шестому",
        "седьмому",
        "восьмому",
        "девятому",
        "десятому"
      ],`

Comment (zotttttttt) [reply locales/ru-RU.json:450]: Russian is too hard to rewrite it well when you have just one version of a word in one particular Case (`Падеж`) :)

File: locales/ru-RU.json (hunk 1 above)
Comment (yuryleb) [reply locales/ru-RU.json:39]: What GPS do you treat as 'standard'? 😉 
But maybe 'Держитесь' will be better, 'Двигайтесь' is just more generic.

File: locales/ru-RU.json (hunk 5 above)
Comment (zotttttttt) [reply locales/ru-RU.json:135]: Thank you, I'll fix it. And "Продолжите" really feels better

File: locales/ru-RU.json (hunk 1 above)
Comment (zotttttttt) [reply locales/ru-RU.json:39]: Yandex.Navigator for me :)

File: locales/ru-RU.json (hunk 5 above)
Comment (zotttttttt) [reply locales/ru-RU.json:135]: Oh no, I misread it. "Продолжите" can be used when you're changing something, there you're just continuing the previous process to some point. Therefore, "Продолжайте" is a better choice

**Comment by kevinkreiser** [comment]
@zotttttttt @yuryleb thank you very much for your work here. i am not well informed about slavic languages however, i would like to mention that if a language needs special functionality in which the replacement tag needs to change based on the grammar of the phrase it is being used in, then we do plan on supporting that in the future. i am well aware of such cases in other languages such as germannic and romance langugaes and i think we can make functionality similar to what is available in OSRM Text Instructions (though likely a different tactic or technique). since we are lacking this ability currently i don't think this is something that should block the current localization efforts here, its just a "bug" or "downside" that we can fix with future fuctionality in my opinion.

please also note that we are not attempting to replace the localization in valhalla with those which are used in OSRM. valhalla has different types of meta data available to its localization engine which we would like to expose in the text strings and this would simply not be possible if we tried that approach.

**Comment by zotttttttt** [comment]
Yes, for sure. I think that every Slavic language would benefit from this new special functionality since they all have so many word forms (e.g., `друг, друга, другу, другом, друге` are the forms of one word in one tense). 

In my take on this localization, I focused more on Valhalla and didn't explore OSRM documentation since these are 2 different projects in my opinion. So, here we have the same thoughts :)

Thank you very much! I'm enjoying this library and would be happy to contribute at least something.

**Review by genadz — APPROVED** [review]

**Review by kevinkreiser — APPROVED** [review]
@genadz who is also a russian speaker shipped this but there was a conflict, so ive resolved that and am just passing his "ship" forward :smile:

**Review by danpaz — APPROVED** [review]
Fixed conflicts and merging.

**Pull request merged**